vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java

```diff
@@ -0,0 +1,294 @@
+        LOGGER.atInfo().addArgument(jsonFilePaths).log("JSON files: {}");
+        return jsonFiles;
+    }
+
+    private static Set<String> getMetaValues(List<Meta> scenarioMeta, String metaName)
+    {
+        return getMetaValuesStream(scenarioMeta, metaName).collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    private static String ensureOneValueOrNull(List<Meta> scenarioMeta, String metaName) throws SyntaxException
+    {
+        List<String> values = getMetaValuesStream(scenarioMeta, metaName).collect(Collectors.toList());
+        if (values.size() > 1)
+        {
+            throw new SyntaxException(String.format("Only one '%s' can be specified for a test case, but got: %s",
+                    metaName, join(values)));
+        }
+        return values.isEmpty() ? null : values.iterator().next();
+    }
+
+    private static String join(Iterable<String> iterable)
+    {
+        return StringUtils.join(iterable, ", ");
+    }
+
+    private static Stream<String> getMetaValuesStream(List<Meta> scenarioMeta, String metaName)
+    {
+        return asStream(scenarioMeta).filter(m -> metaName.equals(m.getName()) && !m.getValue().isEmpty())
+                                     .map(Meta::getValue)
+                                     .map(String::trim)
+                                     .map(value -> StringUtils.splitPreserveAllTokens(value, SEMICOLON))
+                                     .flatMap(Stream::of)
+                                     .map(String::trim);
+    }
+
+    private static boolean isSkipped(List<Meta> scenarioMeta)
+    {
+        return asStream(scenarioMeta).anyMatch(m -> "xray.skip-export".equals(m.getName()));
+    }
+
+    private static Stream<Meta> asStream(List<Meta> scenarioMeta)
+    {
+        return Optional.ofNullable(scenarioMeta)
+                       .map(List::stream)
+                       .orElseGet(Stream::empty);
+    }
+
+    private static final class ErrorExportEntry
+    {
+        private final String story;
+        private final String scenario;
+        private final String error;
+
+        private ErrorExportEntry(String story, String scenario, String error)
+        {
+            this.story = story;
+            this.scenario = scenario;
+            this.error = error;
+        }
+
+        public String getStory()
+        {
+            return story;
+        }
+
+        public String getScenario()
+        {
+            return scenario;
+        }
+
+        public String getError()
+        {
+            return error;
+        }
+    }
+
+    private static final class InvalidIssueStatusException extends Exception
+    {
+        private static final long serialVersionUID = -5547086076322794984L;
+
+        private InvalidIssueStatusException(String testCaseId, String status)
+        {
+            super("Issue " + testCaseId + " is in non-editable '" + status + "' status");
```

Comment:
`Invalid` != `non-editable` - either change default exception message, or pass the full message as constructor parameters, or change exception class name